homeassistant/components/eq3btsmart/climate.py

```diff
@@ -0,0 +1,304 @@
+    """Climate entity to represent a eQ-3 thermostat."""
+
+    _attr_has_entity_name: bool = True
+    _attr_name: str | None = None
+    _attr_supported_features: ClimateEntityFeature = (
+        ClimateEntityFeature.TARGET_TEMPERATURE
+        | ClimateEntityFeature.PRESET_MODE
+        | ClimateEntityFeature.TURN_OFF
+        | ClimateEntityFeature.TURN_ON
+    )
+    _attr_temperature_unit: UnitOfTemperature = UnitOfTemperature.CELSIUS
+    _attr_min_temp: float = EQ3BT_OFF_TEMP
+    _attr_max_temp: float = EQ3BT_MAX_TEMP
+    _attr_precision: float = PRECISION_HALVES
+    _attr_hvac_modes: list[HVACMode] = list(HA_TO_EQ_HVAC.keys())
+    _attr_preset_modes: list[str] = list(Preset)
+    _attr_should_poll: bool = False
+
+    def __init__(self, eq3_config: Eq3Config, thermostat: Thermostat) -> None:
+        """Initialize the climate entity."""
+
+        super().__init__(eq3_config, thermostat)
+
+        self._thermostat.register_update_callback(self._async_on_updated)
+        self._target_temperature: float | None = None
+        self._attr_available = False
+        self._attr_hvac_mode: HVACMode | None = None
+        self._attr_hvac_action: HVACAction | None = None
+        self._attr_preset_mode: str | None = None
+        self._attr_unique_id = format_mac(self._eq3_config.mac_address)
+        self._attr_device_info: DeviceInfo = DeviceInfo(
+            name=self._eq3_config.name,
+            manufacturer=MANUFACTURER,
+            model=DEVICE_MODEL,
+            sw_version=None,
+            serial_number=None,
+            connections={(CONNECTION_BLUETOOTH, self._eq3_config.mac_address)},
+        )
+
+    async def async_added_to_hass(self) -> None:
+        """Run when entity about to be added to hass."""
+
+        async_dispatcher_connect(
+            self.hass, SIGNAL_THERMOSTAT_DISCONNECTED, self._async_on_disconnected
+        )
+        async_dispatcher_connect(
+            self.hass, SIGNAL_THERMOSTAT_CONNECTED, self._async_on_connected
+        )
+
+    @callback
+    def _async_on_disconnected(self, mac_address: str) -> None:
+        if mac_address == self._eq3_config.mac_address:
+            self._attr_available = False
+            self.async_schedule_update_ha_state(force_refresh=True)
+
+    @callback
+    def _async_on_connected(self, mac_address: str) -> None:
+        if mac_address == self._eq3_config.mac_address:
+            self._attr_available = True
+            self.async_schedule_update_ha_state(force_refresh=True)
+
+    @callback
+    def _async_on_updated(self) -> None:
+        """Handle updated data from the thermostat."""
+
+        if self._thermostat.status is not None:
+            self._async_on_status_updated()
+
+        if self._thermostat.device_data is not None:
+            self._async_on_device_updated()
+
+        self.async_schedule_update_ha_state()
+
+    @callback
+    def _async_on_status_updated(self) -> None:
+        """Handle updated status from the thermostat."""
+
```

Comment:
Since you have a single update point. I think it would make this easier to follow if you got rid of `target_temperature` and `current_temperature` and set them as `_attr_target_temperature` and `_attr_current_temperature` here (you may want to convert them to normal functions to keep the return pattern)